hpx/traits/action_schedule_thread.hpp

```diff
@@ -46,6 +47,7 @@ namespace hpx { namespace traits
             {
                 // by default we forward this to the component type
                 typedef typename Action::component_type component_type;
+                data.initial_state = initial_state;
                 component_type::schedule_thread(lva, comptype, data, initial_state);
```

Comment:
While removing the `initial_state` argument here might be a breaking change, I think we should go for it as most probably there will not be a lot of code out there relying on this customization point.